server/aesdsocket.c

```diff
@@ -47,116 +46,463 @@ int main(int argc, char *argv[]) {
+    SLIST_FOREACH(current_entry, &thread_manager, entries) {
+        if (current_entry->thread_id == thread_id) {
+            // set the is_complete field for this thread to true
+            current_entry->is_complete = true;
+
+            // unlock mutex before returning
+            syslog(LOG_DEBUG, "Unlocking manager mutex.");
+            pthread_mutex_unlock(&manager_mutex);
+
+            // return
+            return 0;
+        }
+    }
+    syslog(LOG_DEBUG, "Unlocking manager mutex.");
+    pthread_mutex_unlock(&manager_mutex);
+
+    // no thread_ids matched
+    return -1;
+}
+
+void thread_entry_freeall() {
+    // create a current node to use for checking
+    thread_entry_t *current_entry = NULL;
+
+    while(!(SLIST_EMPTY(&thread_manager))) {
+        // get head of thread manager
+        current_entry = SLIST_FIRST(&thread_manager);
+
+        // join thread
+        pthread_join(current_entry->thread_id, NULL);
+
+        // remove this from thread manager
+        thread_entry_remove(current_entry->thread_id);
+    }
+}
+
+void thread_entry_print(thread_entry_t *current_entry) {
+    // print info
+    syslog(LOG_DEBUG, "Thread ID: %d | Client IP: %s | Client FD: %d | Completion Status: %s\n",
+        (int)current_entry->thread_id, current_entry->client_ip, current_entry->client_fd, current_entry->is_complete ? "Yes" : "No");
+}
+
+void thread_entry_printall() {
+    // create a current node to use for checking
+    thread_entry_t *current_entry = NULL;
+
+    // print header
+    syslog(LOG_DEBUG, "===> THREAD MANAGER HEAD\n");
+
+    // lock the manager
+    syslog(LOG_DEBUG, "Locking manager mutex.");
+    pthread_mutex_lock(&manager_mutex);
+    SLIST_FOREACH(current_entry, &thread_manager, entries) {
+        thread_entry_print(current_entry);
+    }
+    syslog(LOG_DEBUG, "Unlocking manager mutex.");
+    pthread_mutex_unlock(&manager_mutex);
+
+    // print footer
+    syslog(LOG_DEBUG, "===> THREAD MANAGER TAIL\n");
+}
+
+/**************************************************************************************************
+ * FUNCTIONS - TIMESTAMP HANDLER
+ **************************************************************************************************/
+void append_timestamp()
+{
+    // set up variables
+    char timestamp_buffer[64];
+    struct tm *tm_info;
+    time_t current_time;
+
+    // retrieve current time and turn it into RFC 2822 formatted timestamp
+    time(&current_time);
+    tm_info = localtime(&current_time);
+    strftime(timestamp_buffer, sizeof(timestamp_buffer), "timestamp:%a, %d %b %Y %H:%M:%S %z\n", tm_info);
+
+    // open file
+    int timestamp_fd = open(TMPDATA_PATH, O_APPEND | O_RDWR | O_CREAT, S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP);
+    if (timestamp_fd == -1) {
+        syslog(LOG_ERR, "[TIMER] Error opening timestamp file. (errno %d)", errno);
+        return;
+    }
+
+    // write to file
+    syslog(LOG_DEBUG, "Locking tmpdata mutex.");
+    pthread_mutex_lock(&file_mutex);
```

Comment:
@jauy2310 this is unfortunately not safe to do in a signal handler.  Check out the threaded signal handler option to use signals for the timestamp.